src/inferenceql/query/scalar.cljc

```diff
@@ -23,76 +23,96 @@
 
-    [:distribution-event-binop (variable :guard (tree/tag-pred :variable))    [:binop s] (scalar   :guard (tree/tag-pred :scalar-expr))] [(keyword s) (plan variable) (plan scalar)]
-    [:distribution-event-binop (scalar   :guard (tree/tag-pred :scalar-expr)) [:binop s] (variable :guard (tree/tag-pred :variable))]    [(keyword s) (plan variable) (plan scalar)]
+      [:density-event child] (plan child)
+      [:density-event-and & children] (into {} (comp (filter tree/branch?) (map plan)) children)
 
-    [:distribution-event-group "(" child ")"] (plan child)
+      [:density-event-eq (variable :guard (tree/tag-pred :variable)) _= (scalar :guard (tree/tag-pred :scalar-expr))] {(plan variable) (plan scalar)}
+      [:density-event-eq (scalar :guard (tree/tag-pred :scalar-expr)) _= (variable :guard (tree/tag-pred :variable))] {(plan variable) (plan scalar)}
 
-    [:density-event child] (plan child)
-    [:density-event-and & children] (into {} (comp (filter tree/branch?) (map plan)) children)
+      [:density-event-group "(" child ")"] (plan child)
 
-    [:density-event-eq (variable :guard (tree/tag-pred :variable))    _= (scalar   :guard (tree/tag-pred :scalar-expr))] {(plan variable) (plan scalar)}
-    [:density-event-eq (scalar   :guard (tree/tag-pred :scalar-expr)) _= (variable :guard (tree/tag-pred :variable))]    {(plan variable) (plan scalar)}
+      [:probability-expr _prob _of event _under model] `(~'iql/prob ~(plan model) ~(plan event))
+      [:density-expr _prob _density _of event _under model] `(~'iql/pdf ~(plan model) ~(plan event))
 
-    [:density-event-group "(" child ")"] (plan child)
+      [:mutual-info-expr _m _i _of lhs _with rhs _under model] `(~'iql/mutual-info ~(plan model) ~(vec (plan lhs)) ~(vec (plan rhs)))
+      [:approx-mutual-info-expr _a _m _i _of lhs _with rhs _under model] `(~'iql/approx-mutual-info ~(plan model) ~(vec (plan lhs)) ~(vec (plan rhs)))
 
-    [:probability-expr _prob          _of event _under model] `(~'iql/prob ~(plan model) ~(plan event))
-    [:density-expr     _prob _density _of event _under model] `(~'iql/pdf  ~(plan model) ~(plan event))
+      [:model-expr child] (plan child)
+      [:model-expr "(" child ")"] (plan child)
 
-    [:mutual-info-expr           _m _i _of lhs _with rhs _under model] `(~'iql/mutual-info        ~(plan model) ~(vec (plan lhs)) ~(vec (plan rhs)))
-    [:approx-mutual-info-expr _a _m _i _of lhs _with rhs _under model] `(~'iql/approx-mutual-info ~(plan model) ~(vec (plan lhs)) ~(vec (plan rhs)))
+      #?@(:clj [[:generative-table-expr _generative _table relation]
+                (let [query-plan (requiring-resolve 'inferenceql.query.plan/plan)]
+                  `(~'iql/eval-relation-plan (~'quote ~(query-plan relation))))])
 
-    [:model-expr child] (plan child)
-    [:model-expr "(" child ")"] (plan child)
+      [(:or :conditioned-by-expr :conditioned-by-except-clause) & _] (conditioned-by-plan* ws-free-node)
```

Comment:
I am not sure what's going on here. Why is this phrase starting with an `:or`? Maybe add a comment.

Reply:
I'll add a comment.

In `core.match`, `(:or :alternative1 :alternative2)` means it could match either of two alternatives in that spot.

I did that to match either subpart, and delegate further matching/planning to `conditioned-by-plan*`.

And I started delegating to submatching fns because once the core.match patterns got too big/nested, the Clojure compiler started generating invalid Java class names. See https://clojure.atlassian.net/browse/CLJ-1852 and https://ask.clojure.org/index.php/3824/clojure-generated-class-names-length-exceed-file-system-limit?show=3824#q3824 for more.